relay-server/src/actors/global_config.rs

```diff
@@ -254,23 +258,43 @@ impl Service for GlobalConfigService {
             let mut shutdown_handle = Controller::shutdown_handle();
 
             relay_log::info!("global config service starting");
-
-            if self.config.has_credentials() {
-                // NOTE(iker): if this first request fails it's possible the default
-                // global config is forwarded. This is not ideal, but we accept it
-                // for now.
-                self.update_global_config();
-            } else {
-                // NOTE(iker): not making a request results in the sleep handler
-                // not being reset, so no new requests are made.
-                relay_log::info!("fetching global configs disabled: no credentials configured");
-            }
+            match self.config.relay_mode() {
+                RelayMode::Managed => {
+                    if self.config.has_credentials() {
+                        relay_log::info!("serving global configs fetched from upstream");
+                        self.request_global_config();
```

Comment:
btw, would you think its bad that before the first response comes, that the default global config will be served? are there cases where we would rather spool envelopes until it arrives?

Reply:
That's an interesting edge case. Technically it's possible that it happens, but it should be unlikely so we accepted it when we discussed the state machine in the processor. I documented this limitation [here (internal)](https://www.notion.so/sentry/Limitations-of-the-global-config-service-611c9f3700794c5a8e0255512f74be81?pvs=4#0637008d6aa248f9849e85bfb6ddb850).